### PR TITLE
32 update side nav on details page

### LINKED ISSUE #32: Update side navigation on the #docs and #configuration tabs

### PR DESCRIPTION
**Pull request by solazio**
## Done

- Add vanilla side navigation to `#docs` and `#configurations` pages

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8045
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- [List additional steps to QA the new features or prove the bug has been resolved]

## Issue / Card

Fixes #32

### CODE REVIEW

**Comment by spencerbygraves** [comment]
@solazio can we leave one empty column between the left nav and the content please, and one after the content.

image

Thanks

File: static/sass/_pattern_p-side-navigation.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      }
+
+      &:hover {
+        .p-side-navigation__status {
+          display: block;
+        }
+      }
+
+      &.is-active {
+        @include vf-highlight-bar($color-sidenav-item-border-highlight, left);
```
Comment (anthonydillon) [inline static/sass/_pattern_p-side-navigation.scss:23]: You should need to do this. This highlight should use the built in accent color. https://github.com/canonical-web-and-design/vanilla-framework/issues/2966

File: static/sass/styles.scss [hunk 2]
```diff
@@ -5,11 +5,15 @@ $brand-color: #044f66;
 // navigation settings
 $theme-default-nav: dark;
 
+// side navigation settings
+$color-sidenav-item-background-highlight: #cef3ff;
+$color-sidenav-item-border-highlight: #35cdff;
```
Comment (anthonydillon) [inline static/sass/styles.scss:10]: You have these defined in the pattern. You shouldn't need this here.

File: static/sass/styles.scss [hunk 3]
```diff
@@ -5,11 +5,15 @@ $brand-color: #044f66;
...
+
 // import settings
 
 // import custom mixins
 
-// import cookie policy
+// import cookie policys
```
Comment (anthonydillon) [inline static/sass/styles.scss:16]: I don't think this comment is needed. It's clear from the import itself.

File: static/sass/styles.scss [hunk 4]
```diff
@@ -28,12 +32,14 @@ $theme-default-nav: dark;
...
 @include vf-p-links;
 @include vf-p-lists;
 @include vf-p-media-object;
 @include vf-p-muted-heading;
 @include vf-p-navigation;
 @include vf-p-strip;
+@include vf-p-side-navigation;
 @include vf-p-table-of-contents;
 @include vf-p-table-mobile-card;
 @include vf-p-tabs;
```
Comment (anthonydillon) [inline static/sass/styles.scss:45]: I wonder at this point would you be better off including the entire of Vanilla?

File: static/sass/styles.scss [hunk 5]
```diff
@@ -104,6 +106,20 @@ $theme-default-nav: dark;
...
+// https://github.com/canonical-web-and-design/vanilla-framework/issues/2960
+.p-side-navigation__link {
+  .p-side-navigation__status.is-toggle {
+    display: none;
+  }
+
+  &:hover {
+    .p-side-navigation__status.is-toggle {
+      display: block;
+    }
```
Comment (anthonydillon) [inline static/sass/styles.scss:119]: This is in the duplicate, no?

**Comment by spencerbygraves** [comment]
@solazio thanks for updating.

**Pull request merged**